test/fixtures/es-module-loaders/mock-loader.mjs

```diff
@@ -54,127 +43,30 @@ function onPreloadPortMessage({
-      let namespace = Object.create(null);
-      /**
-       * @type {Array<(name: string)=>void>} functions to call whenever an
-       *                                     export name is updated
-       */
-      let listeners = [];
-      for (const name of exportNames) {
-        let currentValueForPropertyName = replacementProperties[name];
-        Object.defineProperty(namespace, name, {
-          enumerable: true,
-          get() {
-            return currentValueForPropertyName;
-          },
-          set(v) {
-            currentValueForPropertyName = v;
-            for (let fn of listeners) {
-              try {
-                fn(name);
-              } catch {
-              }
-            }
-          }
-        });
-      }
-      mockedModules.set(resolved, {
-        namespace,
-        listeners
-      });
-      mockVersion++;
-      // Inform the loader that the `resolved` URL should now use the specific
-      // `mockVersion` and has export names of `exportNames`
-      //
-      // This allows the loader to generate a fake module for that version
-      // and names the next time it resolves a specifier to equal `resolved`
-      port.postMessage({ mockVersion, resolved, exports: exportNames });
-      return namespace;
-    }
-    // Sets the import.meta properties up
-    // has the normal chaining workflow with `defaultImportMetaInitializer`
-    setImportMetaCallback((meta, context, defaultImportMetaInitializer) => {
-      /**
-       * 'node:mock' creates its default export by plucking off of import.meta
-       * and must do so in order to get the communications channel from inside
-       * preloadCode
-       */
-      if (context.url === 'node:mock') {
-        meta.doMock = doMock;
-        return;
-      }
-      /**
-       * Fake modules created by `node:mock` get their meta.mock utility set
-       * to the corresponding value keyed off `mockedModules` and use this
-       * to setup their exports/listeners properly
-       */
-      if (context.url.startsWith('mock-facade:')) {
-        let [proto, version, encodedTargetURL] = context.url.split(':');
-        let decodedTargetURL = decodeURIComponent(encodedTargetURL);
-        if (mockedModules.has(decodedTargetURL)) {
-          meta.mock = mockedModules.get(decodedTargetURL);
-          return;
-        }
-      }
-      /**
-       * Ensure we still get things like `import.meta.url`
-       */
-      defaultImportMetaInitializer(meta, context);
-    });
-  };
-  return `(${insideAppContext})(getBuiltin, port, setImportMetaCallback)`
+/** @type {URL['href']} */
+let mainImportURL
+/** @type {MessagePort} */
+let preloadPort;
+export async function initialize(data) {
+  ({ mainImportURL, port: preloadPort } = data);
+  
+  data.port.on('message', onPreloadPortMessage);
 }
 
+/**
+ * FIXME: this is a hack to workaround loaders being
+ * single threaded for now, just ensures that the MessagePort drains
```

Comment:
@aduh95 This comment isn't correct anymore.

Reply:
```suggestion
 * Because Node.js internals use a separate MessagePort for cross-thread
 * communication, there could be some messages pending that we should handle
 * before continuing.
```